ochazuke/helpers.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Some helpers for the data processing section."""
+
+import datetime
+
+
+def get_days(from_date, to_date):
+    """Create the list of dates spanning two dates.
+
+    A date is a string 'YYYY-MM-DD'
+    A date is considered to be starting at 00:00:00.
+    An invalid date format should be ignored and return None.
+    The same from_date and to_date should return from_date.
+    """
+    date_format = '%Y-%m-%d'
+    try:
+        start = datetime.datetime.strptime(from_date, date_format)
+        end = datetime.datetime.strptime(to_date, date_format)
+        # we assume that the person is requesting one day
+        if start == end:
+            return [from_date]
+    except Exception:
+        return None
+    else:
+        dates = []
+        delta = end - start
+        days = delta.days
+        if days < 0:
+            end = start
+            days = abs(days)
+        for n in range(1, days+1):
+            new_date = end - datetime.timedelta(days=n)
+            dates.append(new_date.strftime(date_format))
+    return dates
```

Comment:
Do we ever care about order? I'm not sure what this would mean... should we enforce that from comes before to? Or maybe the UI doesn't know or care and it's fine to be undefined.

```
>>> get_days('2018-10-5', '2018-10-01')
['2018-10-04', '2018-10-03', '2018-10-02', '2018-10-01']
>>> get_days('2018-10-1', '2018-10-05')
['2018-10-04', '2018-10-03', '2018-10-02', '2018-10-01']
```

Reply:
it accepts both orders,
see the two tests in https://github.com/webcompat/webcompat-metrics-server/pull/25/files#diff-0cda7eb27a25c52683bc500457ee25bfR105

Reply:
also if any mistakes in any parameters, invalid, etc. instead of failing, this is (for now) just sending back the full dataset.